src/components/scenes/FioRequestListScene.js

```diff
@@ -0,0 +1,481 @@
+// @flow
+
+import React, { Component } from 'react'
+import { Alert, Animated, FlatList, Image, Platform, TouchableHighlight, TouchableOpacity, View } from 'react-native'
+import { Actions } from 'react-native-router-flux'
+import slowlog from 'react-native-slowlog'
+
+import checkedIcon from '../../assets/images/createWallet/check_icon_lg.png'
+import invalidIcon from '../../assets/images/createWallet/invalid_icon.png'
+import fioRequestsIcon from '../../assets/images/fio/fio_sent_request.png'
+import * as Constants from '../../constants/indexConstants'
+import { intl } from '../../locales/intl'
+import s from '../../locales/strings.js'
+import FullScreenLoader from '../../modules/FioRequest/components/FullScreenLoader'
+import SwipeListView from '../../modules/FioRequest/components/SwipeListView'
+import { isRejectedFioRequest, isSentFioRequest } from '../../modules/FioRequest/util'
+import T from '../../modules/UI/components/FormattedText/index'
+import { styles as requestListStyles } from '../../styles/scenes/FioRequestListStyle'
+import styles from '../../styles/scenes/TransactionListStyle'
+import type { GuiWallet } from '../../types/types'
+import { SceneWrapper } from '../common/SceneWrapper'
+import { showError } from '../services/AirshipInstance'
+
+const SCROLL_THRESHOLD = 0.5
+
+export type State = {
+  rejectLoading: boolean
+}
+
+export type StateProps = {
+  wallets: { [walletId: string]: GuiWallet },
+  loading: boolean,
+  isoFiatCurrencyCode: string,
+  fiatSymbol: string,
+  exchangeRates: any,
+  pendingFioRequests: any[],
+  sentFioRequests: any[],
+  animation: any,
+  isConnected: boolean
+}
+
+export type DispatchProps = {
+  updateExchangeRates: () => any,
+  getFioRequestsPending: () => any,
+  getFioRequestsSent: () => any,
+  fioRejectRequest: (fioRequestId: string, payerFioAddress: string, cb: Function) => void,
+  removeFioPendingRequest: (requestId: string) => any,
+  setFioPendingRequestSelected: (obj: Object) => any,
+  setFioSentRequestSelected: (obj: Object) => any
+}
+
+type Props = StateProps & DispatchProps
+
+const PENDING_TEXT = s.strings.fio_pendingrequest
+const SENT_TEXT = s.strings.fio_sentrequest
+
+export class FioRequestList extends Component<Props, State> {
+  underlayColor = '#AAA'
+  listFooterHeight = 50
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      rejectLoading: false
+    }
+    slowlog(this, /.*/, global.slowlogOptions)
+  }
+
+  componentDidMount = () => {
+    this.props.updateExchangeRates()
```

Comment:
These three do not have to be redux dispatch calls. They can simply be async operations local to the component that call setstate to change component state once the calls return